src/actions/fee-models/CrossRatePriceFeed.sol

```diff
@@ -0,0 +1,154 @@
+/// - Cross Rate: USDC/ETH = (USDC/USD) / (ETH/USD) = 1/3000 = 0.000333...
+///
+/// The formula is: (numerator / denominator) to get the rate of numerator in terms of denominator
+contract CrossRatePriceFeed is ICrossRatePriceFeed {
+    AggregatorV3Interface public immutable NUMERATOR_FEED; // The feed in the numerator (e.g., USDC/USD)
+    AggregatorV3Interface public immutable DENOMINATOR_FEED; // The feed in the denominator (e.g., ETH/USD)
+
+    uint8 public immutable DECIMALS;
+    uint256 public immutable VERSION;
+    string public description;
+
+    /// @notice Constructor for CrossRatePriceFeed
+    /// @param _numeratorFeed The price feed for the numerator (e.g., USDC/USD)
+    /// @param _denominatorFeed The price feed for the denominator (e.g., ETH/USD)
+    constructor(address _numeratorFeed, address _denominatorFeed) {
+        NUMERATOR_FEED = AggregatorV3Interface(_numeratorFeed);
+        DENOMINATOR_FEED = AggregatorV3Interface(_denominatorFeed);
+
+        // Use the higher precision for calculations to maintain accuracy
+        uint8 numeratorDecimals = NUMERATOR_FEED.decimals();
+        uint8 denominatorDecimals = DENOMINATOR_FEED.decimals();
+        DECIMALS = numeratorDecimals > denominatorDecimals ? numeratorDecimals : denominatorDecimals;
+
+        // Create cross-rate description showing the full formula: "(numerator) / (denominator)"
+        description =
+            string(abi.encodePacked("(", NUMERATOR_FEED.description(), ") / (", DENOMINATOR_FEED.description(), ")"));
+        VERSION = 1;
+    }
+
+    /// @notice Get the cross-rate for a specific round
+    /// @param _roundId The round ID to query
+    /// @return roundId The round ID
+    /// @return answer The cross-rate answer
+    /// @return startedAt When the round started
+    /// @return updatedAt When the round was updated
+    /// @return answeredInRound The round in which the answer was computed
+    function getRoundData(uint80 _roundId)
+        external
+        view
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        int256 numeratorAnswer;
+        int256 denominatorAnswer;
+
+        // Get data from numerator feed (e.g., USDC/USD)
+        (roundId, numeratorAnswer, startedAt, updatedAt, answeredInRound) = NUMERATOR_FEED.getRoundData(_roundId);
+
+        // Get timestamp data from denominator feed to use the latest
+        (,, uint256 denominatorStartedAt, uint256 denominatorUpdatedAt,) = DENOMINATOR_FEED.getRoundData(_roundId);
+
+        // Use the later timestamps to ensure both feeds have data
+        if (denominatorUpdatedAt > updatedAt) {
+            updatedAt = denominatorUpdatedAt;
+        }
+        if (denominatorStartedAt > startedAt) {
+            startedAt = denominatorStartedAt;
+        }
+
+        // Get the answer from denominator feed for the same round
+        (, denominatorAnswer,,,) = DENOMINATOR_FEED.getRoundData(_roundId);
```

Comment:
Hmm may be nbd tho seems like this assumes round IDs are synchronized across feeds even tho it's possible one feed updates more frequently than the other. Is it possible to just use the latest round data for both feeds since we're really just trying to compare the freshest values...